### DISCUSSION TITLE
How to configure heterogeneous multi-programmed simulation?

### DISCUSSION THREAD

**Discussion by b04505009** [CMU-SAFARI/Hermes#8] [General]
How to configure heterogeneous multi-programmed simulation and reproduce on my local machine?

**Comment by rahulbera** [comment]
To run multi-programmed runs, you first need to create a trace list. You can use both homogeneous (meaning multiple copies of the same trace is running across different cores) and heterogeneous (meaning each core runs a different trace) trace mixing strategy. Here is an example of how a four-core trace list might look: https://github.com/CMU-SAFARI/Pythia/blob/master/experiments/MICRO21_4C.tlist. Please compile ChampSim with the desired number of cores and run the same experiment file with the newly-generated executable and trace list. Please note that multi-core runs might take a long time (~14 hours) to finish, depending on the warmup/simulation length.
